subsys/fs/ext2/ext2_impl.c

```diff
@@ -0,0 +1,1558 @@
+/*
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/fs/fs.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+
+#include "ext2.h"
+#include "ext2_impl.h"
+#include "ext2_struct.h"
+#include "ext2_diskops.h"
+#include "ext2_bitmap.h"
+
+LOG_MODULE_REGISTER(ext2, CONFIG_EXT2_LOG_LEVEL);
+
+static struct ext2_data __fs;
+static bool initialized;
+
+/* Structures for blocks slab alocator */
+struct k_mem_slab ext2_block_memory_slab, ext2_block_struct_slab;
+char __aligned(8) __ext2_block_memory_buffer[CONFIG_EXT2_BLOCKS_BUF_SIZE];
+char __aligned(8) __ext2_block_struct_buffer[CONFIG_EXT2_BLOCKS_BUF_SIZE / 64];
```

Comment:
I didn't want to use `K_MEM_SLAB_DEFINE` because it statically initializes it. I need initialize it dynamically when we know the `block_size` that will be used. (That initialization is done in `ext2_init_blocks_slab` function.)